plugins/functions/string.c

```diff
@@ -294,6 +304,165 @@ func_string_tokenize(grn_ctx *ctx, int n_args, grn_obj **args,
   return tokens;
 }
 
+#ifdef GRN_WITH_ONIGMO
+static grn_obj *
+string_regex_slice(grn_ctx *ctx, int n_args, grn_obj **args, grn_user_data *user_data)
+{
+  grn_obj *target, *pattern, *capture, *default_value, *result;
+  int64_t target_length, capture_name_length, capture_num = -1;
+  OnigRegex regex = NULL;
+  OnigPosition position, start, end;
+  OnigRegion *region;
+  const char *capture_name, *target_string;
+
+  if (!(n_args == 3 || n_args == 4)) {
+    GRN_PLUGIN_ERROR(ctx, GRN_INVALID_ARGUMENT,
+                     "[regex_slice] "
+                     "wrong number of arguments (%d for 3...4)",
+                     n_args);
+    return NULL;
+  }
+
+  target = args[0];
+  pattern = args[1];
+  capture = args[2];
+  if (n_args == 4) {
+    default_value = args[3];
+  } else {
+    default_value = grn_plugin_proc_alloc(ctx, user_data, target->header.domain, 0);
+    if (!default_value) {
+      return NULL;
+    }
+
+    GRN_BULK_REWIND(default_value);
+  }
+
+  if (!grn_obj_is_text_family_bulk(ctx, target)) {
+    grn_obj inspected;
+
+    GRN_TEXT_INIT(&inspected, 0);
+    grn_inspect(ctx, &inspected, target);
+    GRN_PLUGIN_ERROR(ctx, GRN_INVALID_ARGUMENT,
+                     "[regex_slice][target] must be a text bulk: <%.*s>",
+                     (int)GRN_TEXT_LEN(&inspected),
+                     GRN_TEXT_VALUE(&inspected));
+    GRN_OBJ_FIN(ctx, &inspected);
+    return NULL;
+  }
+
+  if (!grn_obj_is_text_family_bulk(ctx, capture) && !grn_obj_is_number_family_bulk(ctx, capture)) {
+    grn_obj inspected;
+
+    GRN_TEXT_INIT(&inspected, 0);
+    grn_inspect(ctx, &inspected, capture);
+    GRN_PLUGIN_ERROR(ctx, GRN_INVALID_ARGUMENT,
+                     "[regex_slice][capture] must be a text or number bulk: <%.*s>",
```

Comment:
We don't need `<...>` for inspected value:

```suggestion
                     "[regex_slice][capture] must be a text or number bulk: %.*s",
```

`<...>` is for highlighting invisible values such as ` `. But inspected string is surrounded by `"..."`. So we don't need extra `<...>`.